env.vs2017_xp.bat

```diff
@@ -6,10 +6,18 @@ rem REQUIRED: path to Boost source directory
 if not defined BOOST_ROOT set BOOST_ROOT=%RIME_ROOT%\deps\boost_1_69_0
 
 rem architecture, Visual Studio version and platform toolset
-set ARCH=Win32
-set BJAM_TOOLSET=msvc-14.1
-set CMAKE_GENERATOR="Visual Studio 15 2017"
-set PLATFORM_TOOLSET=v141_xp
+if not defined CMAKE_GENERATOR (
+    set ARCH=Win32
+    set BJAM_TOOLSET=msvc-14.1
+    set CMAKE_GENERATOR="Visual Studio 15 2017"
+    set PLATFORM_TOOLSET=v141_xp
+) else (
+    set "VSVEROPT=-version [15.0^,16.0^)"

```

Comment:
If `CMAKE_GENERATOR` is defined outside the scripts, it may not be `Visual Studio xx xxxx`. To preserve the original settings, I split out a else-block for non-Visual-Studio generator to run a script which call `vswhere.exe` to find the Visual Studio's install location and call `vcvarsall.bat` to set almost all environment variables we need.